lib/memory.gi

```diff
@@ -147,6 +147,9 @@ InstallGlobalFunction( SLPOfElms,
   function(elms)
     # Returns a straight line program to write elm as in the original 
     # generators.
+    if ForAny(elms{[2..Length(elms)]}, x -> not IsIdenticalObj(elms[1]!.slp, x!.slp)) then
+        ErrorNoReturn("SLPOfElms: the slp components of all elements must be identical");
+    fi;
```

Comment:
Perhaps add a test to get coverage on this?

it seems there are no test for this whole function, though

Reply:
I wasn't sure whether there are tests for this functions. I'll make sure to add some.

Reply:
I have added some tests.

Reply:
Nice, looks good to me.